dockerized front & back ends, added docker compose

SeanEZhou opened the pull request.

McSobkov reviewed (commented):
Not sure how to tag a specific file, but curious as to why you changed the console.log to localhost in routes.js. Seems to me with deployment it might be even more helpful to display the actual hostname. Docker stuff looks solid though. Should we run the test scripts too or are those not necessary since git takes care of those?

SeanEZhou commented:
> Not sure how to tag a specific file, but curious as to why you changed the console.log to localhost in routes.js. Seems to me with deployment it might be even more helpful to display the actual hostname. Docker stuff looks solid though. Should we run the test scripts too or are those not necessary since git takes care of those?

I changed that because the local host is no longer 127.0.0.1 but the docker container that runs the backend. Notice how I deleted the hostname 127.0.0.1 in app.listen to make it work.
